File: src/com/nimbusly/isometrix/demo/TextureDrawBackgroundDemo.java
package com.nimbusly.isometrix.demo;

import javax.microedition.khronos.opengles.GL10;

import android.content.Context;

import com.nimbusly.isometrix.Background;
import com.nimbusly.isometrix.GameRenderer;

public class TextureDrawBackgroundDemo extends GameRenderer {
	

	private static final String TAG = "TestRenderer";
	private Background background;
	int vx = 3, vy = 2;
	
	public TextureDrawBackgroundDemo(Context context) {
		super(context);
	}
	
	private int bounce(int x, int vx, int max) {
		return (x > max || x < 0) ? -vx : vx;
	}
	
	public void onDrawFrame(GL10 gl) {
		super.onDrawFrame(gl);
		background.x += vx; background.y += vy;
		vx = bounce(background.x, vx, background.textureWidth-background.viewWidth);
		vy = bounce(background.y, vy, background.textureHeight-background.viewWidth);
		background.draw(gl);
	}

	public void onSurfaceChanged(GL10 gl, int width, int height) {
		super.onSurfaceChanged(gl, width, height);
		background = new Background(context, gl, R.drawable.amelia, width, height);
	}
	
}

File: src/com/nimbusly/isometrix/demo/TextureDrawBackgroundDemo.scala
package com.nimbusly.isometrix.demo

import javax.microedition.khronos.opengles.GL10
import android.content.Context
import com.nimbusly.isometrix.Background
import com.nimbusly.isometrix.GameRenderer

class TextureDrawBackgroundDemo(context: Context) extends GameRenderer(context) {

	private val TAG = "TextureDrawBackgroundDemo"
	private var background: Background = null
	var vx = 3
	var vy = 2
	
	private def bounce(x: Int, vx: Int, max: Int) = if (x > max || x < 0) -vx else vx
	
	override def onDrawFrame(gl: GL10) {
		super.onDrawFrame(gl)
		background.x += vx 
		background.y += vy
		vx = bounce(background.x, vx, background.textureWidth-background.viewWidth)
		vy = bounce(background.y, vy, background.textureHeight-background.viewWidth)
		background.draw(gl)
	}

	override def onSurfaceChanged(gl: GL10, width: Int, height: Int) {
		super.onSurfaceChanged(gl, width, height)
		background = new Background(context, gl, R.drawable.amelia, width, height)
	}
}